src/block.rs

```diff
@@ -85,34 +79,23 @@ where
     raw_decode::<C, H, O, D>(cid.codec(), data)
 }
 
-/// Decode raw ipld.
-///
-/// Useful for nested encodings when for example the data is encrypted.
-pub fn raw_decode_ipld(codec: CCode, data: &[u8]) -> Result<Ipld> {
-    match codec {
```

Comment:
Can we keep this? If the data is encrypted the cid hash won't match, as the cid matches the encrypted data

Reply:
Have you seen the intermediate step before it was removed at https://github.com/ipfs-rust/rust-ipld/pull/45/commits/757f94cf43b8ebc9759f07d8e7976e39a3619d51 ?

The `raw_decode_ipld()` call was reduced to basically `codec.decode(data)`. Hence I thought about removing it, as it is easy enough to call that directly. Would that work for your use case?

Reply:
I see, makes sense